Removed the Accordion "summary" role="tab" to improve accessibility

piopi opened the pull request:
Removed the Accordion "summary" role="tab" to improve the accessibility, since ARIA role tab is not allowed for summary tags. Please see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary for more details.  Fixes #8919.

Commit:
Removed the Accordion "summary" role="tab" to improve the accessibility, since ARIA role tab is not allowed for summary. Please see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary for more details.  Fixes #8919.

Commit:
Edited the tests to reflect, the removal of the aria role for summary tag

duboisp reviewed (dismissed):
The solution would need to impact the tabs interface too because it use that pattern.

piopi commented:
@duboisp thank you for your feedback. I checked the WCAG documentation https://www.w3.org/TR/html-aria/, and we can see that the summary and details tags cannot have defined roles because they have an implicit role of a button and a group respectively and also have some additional aria-* attributes restrictions. As for the wrapping, they are wrapped actually in a tablist elements: https://github.com/wet-boew/wet-boew/blob/0ccb73de71e284cd2b486264f237972a2071b660/src/plugins/toggle/toggle.js#L97
The issue with a tablist elements is the requirement that all of its children need to have a role of tab and since we can't give roles to the summary/details tags, I can see two solutions for this problem. 
1. Wrap the details/summary in a div that has a tab role and the appropriate aria-* attributes while keeping the tablist role on the parent

or

2. Remove the tablist role from the parents and make the appropriate changes on the tabs interface

I will start with the first solution and do another PR.

Commit:
Wrapped the details/summary in a div, having the appropriate aria roles and attributes.

piopi commented:
Added the div wrapping, removed the incompatible aria role/attributes from the summary/details accordion, and made sure to give the div the appropriate aria-roles and attributes.

As we can see from the screenshots below, these changes removed the accessibility issues found before:

**Before**
<details><summary>Click to expand </summary>

![image](https://user-images.githubusercontent.com/23510615/95029266-fd440b00-0674-11eb-859a-f90e33598238.png)

![image](https://user-images.githubusercontent.com/23510615/95029315-4bf1a500-0675-11eb-8cc7-23ef23f8232f.png)

</details>

**After**
<details><summary>Click to expand </summary>

![image](https://user-images.githubusercontent.com/23510615/95029272-07fea000-0675-11eb-8720-af4278ce0bba.png)

![image](https://user-images.githubusercontent.com/23510615/95029353-983ce500-0675-11eb-99ce-f7e4171d0f3e.png)

</details>

piopi commented:
After testing with NVDA screen reader v.2020.2 on Firefox and Chrome, it seems that the issue #8918 is also solved with this solution.
![image](https://user-images.githubusercontent.com/23510615/95035724-f2e53980-0693-11eb-9cc4-17450fa802bf.png)

duboisp commented:
Change approved upon successful testing.

duboisp commented:
@piopi and @GormFrank we will need to revert those change, it prevent to initialize the subsequent tabbed interface.

You can see the issue here: https://wet-boew.github.io/wet-boew/demos/tabs/tabs-en.html and the error console show an error "Uncaught DOMException: Node.replaceChild: Child to be replaced is not a child of this node" for line 131 - https://github.com/wet-boew/wet-boew/blame/master/src/plugins/toggle/toggle.js#L131

piopi commented:
@duboisp @GormFrank Issue was solved here https://github.com/wet-boew/wet-boew/pull/8986
